src/main/java/net/nikr/eve/jeveasset/gui/tabs/skills/SkillsOverviewTab.java

```diff
@@ -0,0 +1,510 @@
+		jTable.setColumnSelectionAllowed(true);
+		TableComparatorChooser<Row> comparatorChooser = TableComparatorChooser.install(jTable, sortedColumns, TableComparatorChooser.MULTIPLE_COLUMN_MOUSE, tableFormat);
+		selectionModel = EventModels.createSelectionModel(filterList);
+		selectionModel.setSelectionMode(ListSelection.MULTIPLE_INTERVAL_SELECTION_DEFENSIVE);
+		jTable.setSelectionModel(selectionModel);
+		installTable(jTable);
+		JScrollPane scrollPane = new JScrollPane(jTable);
+		filterControl = new SkillsOverviewFilterControl(sorted);
+		installTableTool(new SkillsOverviewTableMenu(), tableFormat, comparatorChooser, tableModel, jTable, eventList, Row.class);
+
+		layout.setHorizontalGroup(
+				layout.createParallelGroup()
+						.addComponent(filterControl.getPanel())
+						.addComponent(scrollPane, 0, 0, Short.MAX_VALUE));
+		layout.setVerticalGroup(
+				layout.createSequentialGroup()
+						.addComponent(filterControl.getPanel())
+						.addComponent(scrollPane, 0, 0, Short.MAX_VALUE));
+	}
+
+	@Override
+	public void updateData() {
+		try {
+			eventList.getReadWriteLock().writeLock().lock();
+			eventList.clear();
+			for (EnumTableColumn<Row> col : new ArrayList<>(dynamicColumns)) {
+				tableFormat.removeColumn(col);
+			}
+			dynamicColumns.clear();
+			for (String planName : Settings.get().getSkillPlans().keySet()) {
+				PlanColumn col = new PlanColumn(planName);
+				tableFormat.addColumn(col);
+				dynamicColumns.add(col);
+			}
+			tableModel.fireTableStructureChanged();
+			filterControl.updateColumns(true);
+			Map<String, Map<Integer, Integer>> plans = Settings.get().getSkillPlans();
+			Map<String, Map<Integer, Long>> ownerSp = new LinkedHashMap<>();
+			program.getProfileData().getSkillsEventList().getReadWriteLock().readLock().lock();
+			try {
+				for (MySkill s : program.getProfileData().getSkillsEventList()) {
+					Map<Integer, Long> map = ownerSp.computeIfAbsent(s.getOwnerName(), k -> new LinkedHashMap<>());
+					map.put(s.getTypeID(), s.getSkillpoints());
+				}
+			} finally {
+				program.getProfileData().getSkillsEventList().getReadWriteLock().readLock().unlock();
+			}
+			for (Map.Entry<String, Map<Integer, Long>> entry : ownerSp.entrySet()) {
+				Row row = new Row(entry.getKey());
+				row.setOwnerSkillSp(entry.getValue());
+				// Populate Total SP and Unallocated SP for this character
+				OwnerType ownerType = null;
+				for (OwnerType ot : program.getOwnerTypes()) {
+					if (ot.getOwnerName().equals(entry.getKey())) {
+						ownerType = ot;
+						break;
+					}
+				}
+				if (ownerType != null) {
+					row.setTotalSp(ownerType.getTotalSkillPoints());
+					row.setUnallocatedSp(ownerType.getUnallocatedSkillPoints());
+				}
+				for (Map.Entry<String, Map<Integer, Integer>> plan : plans.entrySet()) {
+					double pct = computePlanPercent(entry.getValue(), plan.getValue());
+					row.planToPercent.put(plan.getKey(), Percent.create(pct));
+				}
+				eventList.add(row);
```

Comment:
This is very bad for performances, as adding to the EventList trigger a lot of stuff (sorting, filtering, GUI updates). It's much faster to add everything to a normal List and then use addAll on the EventList (which group all the changes together). Generally the less operation on the EventList, the better.